src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/MoreMenuPage.java

```diff
@@ -12,17 +12,19 @@
 public class MoreMenuPage extends MoreMenuPageBase {
 
     @FindBy(id = "com.myfitnesspal.android:id/image")
-    ExtendedWebElement userAvatar;
+    private ExtendedWebElement userAvatar;
+
+    @FindBy(xpath = "//*[@resource-id='com.myfitnesspal.android:id/moreRecyclerView']//*[@text='%s']")
+    private ExtendedWebElement moreOption;
 
     public MoreMenuPage(WebDriver driver) {
         super(driver);
     }
 
     @Override
     public boolean isOptionPresent(MoreOptions option) {
-        CommonPageBase commonPage = initPage(getDriver(), CommonPageBase.class);
-        commonPage.SwipeToElementByText(option.getOptionText());
-        return commonPage.isItemByTextPresent(option.getOptionText());
+        swipe(moreOption.format(option.getOptionText(), Direction.VERTICAL, 3, 1000));
+        return moreOption.format(option.getOptionText()).isPresent();
```

Comment:
good, do not forget about timeouts - isPresent(2), you can add it to next commit with test for case 8